### ISSUE TITLE
[Feature Request] Ability to disable centering behavior

### ISSUE THREAD

**Issue by marciot** [snorpey/circlepacker#3]
I was curious how difficult it would be to disable the centering behavior of the circle packing algorithm. The reason for this is that I am somewhat misusing this library, not for packing, but for making sure objects don't overlap in [3D printing application](https://syndaverco.github.io/slicer/). I notice there is a function to "setCenteringPasses(number)" but that the argument is documented as requiring a number equal to at least one. Would it be possible to allow that to be zero, in which case, only overlapping circles would be moved to not overlap,  while others would remain where they are? I have not tried to set this to zero to see whether the behavior already exists.

**Comment by marciot** [comment]
So I was able to accomplish what I wanted by changing a single character in the code [here](https://github.com/snorpey/circlepacker/blob/474f6d3eeada62d889be7a55309412942c2f1367/src/CirclePackWorker.js#L63) and it behaved exactly as I had hoped. It seems to work, but I do not know enough to see whether it may have unintended side effects.

```
if ( typeof message === 'number' && message >= 0 ) {
   circleManager.numberOfCollisionPasses = message;
}
```

Not sure whether you consider this use case to be significant enough to include in the mainline code.

**Comment by snorpey** [comment]
hey @marciot thank you for your interest in this library and for feature suggestion. I think it's a great idea and will take a closer look once I have a bit more time on my hands. I hope this is OK.
thanks again!

**Comment by marciot** [comment]
@snorpey: Thank you for the consideration! No rush on this, but I thought of a possible variant which is slightly more comprehensive.

If it were possible to disable the centering pass on objects but not others, similar to `packer.pinCircle(...);`, this would give the user even more control.

I think maybe each circle could have flag indicating whether:

- Whether it is pinned (i.e. a pinned object will never move, but can affect other objects)
- Whether it will be centered (default) or whether it will only move when overlapping another object

This would give the app developer a lot of control over the behavior of individual objects. In my 3D printing app, I would definitely be able to use the feature to give the user more control over how to arrange objects.

**Comment by snorpey** [comment]
@marciot i just published version [1.1.3](https://github.com/snorpey/circlepacker/releases/tag/v1.1.3). have a look at the [`setCircleCenterPull()`](https://github.com/snorpey/circlepacker#setcirclecenterpullcircleid-boolean) and [`setCenterPull()`](https://github.com/snorpey/circlepacker#setcenterpullboolean) methods. let me know if you have any questions. thanks again for your feedback!

**Comment by marciot** [comment]
@snorpey: I just added the new library to Symple Slicer and it works great! With the new feature set, I was able to allow the user to select multiple objects to pack to the center. Works great! Thank you for being so responsive!

> @marciot i just published version [1.1.3](https://github.com/snorpey/circlepacker/releases/tag/v1.1.3). have a look at the [`setCircleCenterPull()`](https://github.com/snorpey/circlepacker#setcirclecenterpullcircleid-boolean) and [`setCenterPull()`](https://github.com/snorpey/circlepacker#setcenterpullboolean) methods. let me know if you have any questions. thanks again for your feedback!
